cpu/stm32/include/periph/cpu_timer.h

```diff
@@ -44,11 +44,25 @@ extern "C" {
  */
 #define TIM_CHAN(tim, chan) *(&dev(tim)->CCR1 + chan)
 
+static const periph_t periph_timer2 = {
+    #if   defined(RCC_APB1ENR_TIM2EN)
+    .en_reg = &RCC->APB1ENR,
+    .en_mask = RCC_APB1ENR_TIM2EN,
+    #elif defined(RCC_APB1ENR1_TIM2EN)
+    .en_reg = &RCC->APB1ENR1,
+    .en_mask = RCC_APB1ENR1_TIM2EN,
+    #elif defined(RCC_MC_APB1ENSETR_TIM2EN)
+    .en_reg = &RCC->MC_APB1ENSETR,
+    .en_mask = RCC_MC_APB1ENSETR_TIM2EN,
+    #endif
+};
+
 /**
  * @brief   Timer configuration
  */
 typedef struct {
     TIM_TypeDef *dev;       /**< timer device */
+    const periph_t *rcc_dev;
```

Comment:
Is there a better name for this pointer?

Reply:
```suggestion
    const rcc_register_t *rcc_register;
```